settlement/bitflyer/upload.go

```diff
@@ -295,139 +285,154 @@ func createBitflyerRequests(
 	}
 	return set, len(settlementRequests) - 1
 }
 
+// PrepareRequests prepares requests
+func PrepareRequests(
+	ctx context.Context,
+	bitflyerClient bitflyer.Client,
+	txs []settlement.Transaction,
+	excludeLimited bool,
+) (*PreparedTransactions, error) {
+	logger, err := appctx.GetLogger(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	quote, err := bitflyerClient.FetchQuote(ctx, "BAT_JPY", true)
+	if err != nil {
+		return nil, err
+	}
+
+	rate := quote.Rate
+
+	// group by wallet provider id
+	groupedByWalletProviderID := GroupSettlements(&txs)
+	// bat limit
+	probiLimit := altcurrency.BAT.ToProbi(decimal.NewFromFloat32(200000). // start with jpy
+										Div(rate).                      // convert to bat
+										Mul(decimal.NewFromFloat(0.9)). // reduce by an extra 10% if we're paranoid
+										Truncate(8))                    // truncated to satoshis
+	_, transactionBatches, notSubmittedTransactions, numReduced, err := setupSettlementTransactions(
+		groupedByWalletProviderID,
+		probiLimit,
+		excludeLimited,
+	)
+
+	jpyBATRate, _ := rate.Float64()
+	logger.Info().Float64("JPY/BAT", jpyBATRate).Int("batches", len(transactionBatches)).Int("ignored", len(notSubmittedTransactions)).Int("reduced", numReduced).Msg("prepared bf transactions")
+
+	return &PreparedTransactions{
+		AggregateTransactionBatches: transactionBatches,
+		NotSubmittedTransactions:    notSubmittedTransactions,
+	}, err
+}
+
+type PreparedTransactions struct {
+	// goes to bitflyer
+	AggregateTransactionBatches [][]settlement.AggregateTransaction `json:"aggregateTransactionBatches"`
+	// a list of settlements that are not being sent
+	NotSubmittedTransactions []settlement.Transaction `json:"notSubmittedTransactions"`
+}
+
 // IterateRequest iterates requests
 func IterateRequest(
 	ctx context.Context,
 	action string,
 	bitflyerClient bitflyer.Client,
-	bulkPayoutFiles []string,
 	sourceFrom string,
-	excludeLimited bool,
+	prepared PreparedTransactions,
 	dryRun *bitflyer.DryRunOption,
 ) (map[string][]settlement.Transaction, error) {
-
 	logger, err := appctx.GetLogger(ctx)
 	if err != nil {
 		_, logger = logging.SetupLogger(ctx)
 	}
+	transactionBatches := prepared.AggregateTransactionBatches
+	notSubmittedTransactions := prepared.NotSubmittedTransactions
 
 	// for submission to eyeshade
 	submittedTransactions := make(map[string][]settlement.Transaction)
 
+	if len(notSubmittedTransactions) > 0 {
+		submittedTransactions[notSubmittedCategory] = append(
+			submittedTransactions[notSubmittedCategory],
+			notSubmittedTransactions...,
+		)
+	}
+
 	quote, err := bitflyerClient.FetchQuote(ctx, "BAT_JPY", true)
 	if err != nil {
-		return submittedTransactions, err
+		return nil, err
 	}
-
-	for _, bulkPayoutFile := range bulkPayoutFiles {
-		bytes, err := ioutil.ReadFile(bulkPayoutFile)
-		if err != nil {
-			logger.Error().Err(err).Msg("failed to read bulk payout file")
-			return submittedTransactions, err
+	for i, batch := range transactionBatches {
```

Comment:
Maybe add a small comment here saying that this is adding transfer IDs to each request?  Will make it easier to scan.